File: leetcode/python/1207_unique_number_of_occurrences.py
from collections import Counter


class Solution:
    def uniqueOccurrences(self, arr: list[int]) -> bool:
        occurrence = Counter(arr)
        return len(occurrence) == len(set(occurrence.values()))
        
s = Solution()
print(s.uniqueOccurrences([1,2,2,1,1,3]))
print(s.uniqueOccurrences([1,2]))

File: leetcode/rust/src/1207_unique_number_of_occurrences.rs
use std::collections::{HashMap, HashSet};

impl Solution {
    pub fn unique_occurrences(arr: Vec<i32>) -> bool {
        let count = arr.iter().fold(
            HashMap::new(), 
            |mut acc, &x| {
                *acc.entry(x).or_insert(0) += 1;
                acc
            });

        let occurrences = count.values().into_iter().fold(
            HashSet::new(), 
            |mut acc, &x| {
                acc.insert(x);
                acc
            });

        occurrences.len() == count.len()
    }
}